requirements.txt

```diff
@@ -1,2 +1 @@
-nose
-sphinx
+-e .
```

Comment:
I'm not sure if adding a comment in the yaml is supported, basically this change delegates all the heavy lifting to setup.py.

Reply:
Would you mind sharing a link to some background information, I'm eager to learn what's going on behind the scene here.

Reply:
I got the pattern from this [r/python thread](https://www.reddit.com/r/Python/comments/3uzl2a/setuppy_requirementstxt_or_a_combination/cxj4mdn/). As always there are heaps of different ways to do, this seemed like the one that doesn't duplicate the list of packages without needing to read requirements.txt in setup.py. Personally I think this is cleaner since everything is in setup.py where we can handle it using code instead of a txt.